client/request.go

```diff
@@ -234,8 +234,35 @@ func (cli *Client) checkResponseErr(serverResp serverResponse) error {
 		if err := json.Unmarshal(body, &errorResponse); err != nil {
 			return errors.Wrap(err, "Error reading JSON")
```

Comment:
Probably one more follow-up coming for this case, where we could end up with a proxy returning HTML with an invalid content-type header, which may result in;

```
Error reading JSON: invalid character '<' looking for beginning of value
```

Possibly we could consider;

- by default capture something similar to below (include the failure to parse, but also reply with status-code and "text")
- adding a debug log to log the response, so that a user could enable `--debug` to see the problem
- maybe same for `text/html` responses, which are unexpected, but we may not always want to dump the HTML as-is.

Reply:
☝️ I had some WIP changes for both, but not yet complete, so I need to dust them off a bit; thought I'd push this PR already, as this one is already OK to review